server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/json/VacationResponseSerializationTest.scala

```diff
@@ -0,0 +1,73 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.json
+
+import java.time.ZonedDateTime
+
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.james.jmap.json.VacationResponseSerializationTest.VACATION_RESPONSE
+import org.apache.james.jmap.mail.{FromDate, HtmlBody, IsEnabled, Subject, TextBody, ToDate, VacationResponse, VacationResponseId}
+import org.apache.james.mailbox.model.TestId
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.Json
+
+object VacationResponseSerializationTest {
+  private val DATE_TIME_2016 = ZonedDateTime.parse("2016-10-09T08:07:06+07:00[Asia/Vientiane]")
+  private val DATE_TIME_2017 = ZonedDateTime.parse("2017-10-09T08:07:06+07:00[Asia/Vientiane]")
+
+  private val VACATION_RESPONSE_ID: VacationResponseId = VacationResponseId()
+  private val IS_ENABLED: IsEnabled = IsEnabled(true)
+  private val FROM_DATE: Option[FromDate] = Some(FromDate(DATE_TIME_2016))
+  private val TO_DATE: Option[ToDate] = Some(ToDate(DATE_TIME_2017))
+  private val SUBJECT: Option[Subject] = Some(Subject("Hello world"))
+  private val TEXT_BODY: Option[TextBody] = Some(TextBody("text is required when enabled"))
+  private val HTML_BODY: Option[HtmlBody] = Some(HtmlBody("<b>HTML body</b>"))
+
+  val VACATION_RESPONSE: VacationResponse = VacationResponse(
+    id = VACATION_RESPONSE_ID,
+    isEnabled = IS_ENABLED,
+    fromDate = FROM_DATE,
+    toDate = TO_DATE,
+    subject = SUBJECT,
+    textBody = TEXT_BODY,
+    htmlBody = HTML_BODY
+  )
+}
+
+class VacationResponseSerializationTest extends AnyWordSpec with Matchers {
+  "Serialize VacationResponse" should {
+    "succeed" in {
+      val expectedJson: String =
+        """{
+          | "id":"singleton",
+          | "isEnabled":true,
+          | "fromDate":"2016-10-09T08:07:06+07:00[Asia/Vientiane]",
+          | "toDate":"2017-10-09T08:07:06+07:00[Asia/Vientiane]",
```

Comment:
This needs to be expressed with an UTCDate:

https://jmap.io/spec-core.html#the-date-and-utcdate-data-types

```
Where UTCDate is given as a type, it means a Date where the time-offset component MUST be Z (i.e., it must be in UTC time). For example, "2014-10-30T06:12:00Z".
```